src/common/file.c

```diff
@@ -199,7 +199,11 @@ util_file_is_device_dax(const char *path)
 	int fd = os_open(path, O_RDONLY);
 	if (fd < 0) {
 		/* not a problem - 'path' may point to non existent file */
-		/* LOG(4, "!open \"%s\"", path); */
+		if (errno == ENOENT)
+			goto out;
+
+		LOG(4, "!open \"%s\"", path);
+		olderrno = errno;
```

Comment:
There are 2 problems:
1) This function should not return a boolean value, because it can fail. Callers should deal with the 3rd value and propagate error up.
2) We don't need to open a file to check whether it is device dax or not. The code in util_fd_is_device_dax can use "stat" syscall to get what it needs.